cloudformation/ec2-asg.yaml

```diff
@@ -0,0 +1,603 @@
+    Properties:
+      HealthCheckIntervalSeconds: 30
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 3
+      HealthCheckPath: /ping
+      HealthCheckProtocol: HTTP
+      Port: 8080
+      Protocol: HTTP
+      UnhealthyThresholdCount: 5
+      VpcId: !Ref 'VPC'
+  ALBManagementListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref 'ALBManagementTargetGroup'
+      LoadBalancerArn: !Ref 'ApplicationLoadBalancer'
+      Port: '8081'
+      Protocol: HTTP
+  ALBInferenceListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref 'ALBInferenceTargetGroup'
+      LoadBalancerArn: !Ref 'ApplicationLoadBalancer'
+      Port: '8080'
+      Protocol: HTTP
+  TorchServeASG:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    DependsOn: GatewayToInternet
+    Properties:
+      VPCZoneIdentifier: [!Ref PublicSubnetA, !Ref PublicSubnetB]
+      LaunchConfigurationName: !Ref 'LaunchConfig'
+      MinSize: !Ref 'MinNodeNumber'
+      MaxSize: !Ref 'MaxNodeNumber'
+      TargetGroupARNs:
+        - !Ref 'ALBInferenceTargetGroup'
+        - !Ref 'ALBManagementTargetGroup'
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT30M
+        Count: '1'
+    UpdatePolicy:
+      AutoScalingRollingUpdate:
+        MinInstancesInService: '1'
+        MaxBatchSize: '1'
+        PauseTime: PT30M
+        WaitOnResourceSignals: 'true'
+  EFSSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: EFS security group
+      SecurityGroupIngress:
+        - SourceSecurityGroupId: !Ref TorcheServeSecurityGroup
+          FromPort: 2049
+          IpProtocol: tcp
+          ToPort: 2049
+      VpcId: !Ref 'VPC'
+  TorchServeEFS:
+    Type: AWS::EFS::FileSystem
+    Properties:
+      Encrypted: false
+  TorchServeMountTargetA:
+    Type: AWS::EFS::MountTarget
+    Properties:
+      FileSystemId: !Ref TorchServeEFS
+      SecurityGroups:
+        - !Ref EFSSecurityGroup
+      SubnetId: !Ref PublicSubnetA
+  TorchServeMountTargetB:
+    Type: AWS::EFS::MountTarget
+    Properties:
+      FileSystemId: !Ref TorchServeEFS
+      SecurityGroups:
+        - !Ref EFSSecurityGroup
+      SubnetId: !Ref PublicSubnetB
+  LaunchConfig:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    DependsOn: GatewayToInternet
+    Metadata:
+      AWS::CloudFormation::Init:
+        configSets:
+          InstallAndRun:
+            - TorchServeSetup
+            - CloudwatchSetup
+            - CFNSetup
+        TorchServeSetup:
+          files:
+            /etc/torchserve/config.properties:
+               content: !Sub |
+                 inference_address=http://0.0.0.0:8080
+                 management_address=http://0.0.0.0:8081
+                 private_key_file=/etc/torchserve/server.key
```

Comment:
if mgmt and inference addresses set are HTTP then do we need these key and cert config?

Reply:
@dhanainme Good catch. Will remove